### PR TITLE
CliConfig surface, CLI config enums, and OrtConfig upgrade to 0.8.0

### PR DESCRIPTION
**Pull request by leynos**
<pr-body>
## Summary
- Introduce a dedicated, typed CLI configuration surface (CliConfig) derived from OrthoConfig v0.8.0 and extend the CLI surface with new typed fields (ColourPolicy, SpinnerMode, OutputFormat) and default_targets with append merge semantics. Preserve backward compatibility with existing flags/env/file precedence and route resolution through CliConfig where appropriate.
- Add localization hooks and tests scaffolding for the new fields, with updated docs and build helpers to accommodate the shared config surface.
- Upgrade OrtConfig to v0.8.0 in Cargo.toml/build-deps and align build script anchors.
- Provide a dedicated execplan document describing CliConfig design and integration.

## Changes
- New module: src/cli/config.rs
  - ColourPolicy: Auto, Always, Never (derive Serialize/Deserialize, FromStr, Display)
  - SpinnerMode: Enabled, Disabled (derive Serialize/Deserialize, FromStr, Display)
  - OutputFormat: Human, Json (derive Serialize/Deserialize, FromStr, Display)
  - CliConfig: a typed configuration view for CLI/config layers with fields for verbose, locale, accessible, no_emoji, theme, colour_policy, progress, spinner_mode, diag_json, output_format, and default_targets (merge strategy: append).
- Extended CLI surface: src/cli/mod.rs
  - Add fields to Cli:
    - colour_policy: Option<ColourPolicy>
    - spinner_mode: Option<SpinnerMode>
    - output_format: Option<OutputFormat>
    - default_targets: Vec<String> (#[ortho_config(merge_strategy = "append")])
  - Expose a config() accessor to obtain a CliConfig from Cli.
- Merging / configuration: src/cli/config_merge.rs
  - Include the new fields in cli_overrides_from_matches so file/env values aren’t overridden by absent CLI defaults.
- Parsing & validation: src/cli/parsing.rs and src/cli/validation.rs
  - Add parsers for ColourPolicy, SpinnerMode, and OutputFormat; wire them into the validation layer.
- Localization & keys:
  - src/cli_l10n.rs: map new argument IDs to Fluent keys.
  - locales/en-US/es-ES: new keys for the new flags and validations.
- Documentation & design:
  - Update docs to describe CliConfig and new fields; add localization and test scaffolding notes.
- Artifacts and tests:
  - Build script anchors updated to handle new shared helpers.
  - Tests scaffolding for unit tests (rstest) and extended CLI merge/parsing tests; BDD scaffolding for cli_config feature.
  - Added new tests and feature files under tests/ and tests/features/cli_config.feature.
- Build & dependencies:
  - OrtConfig upgraded from 0.7.0 to 0.8.0 in Cargo.toml and build-deps.
- Docs:
  - netsuke-design, users-guide, and roadmap references updated to reflect CliConfig introduction.

## Why this approach
- Keeps Cli as the single source of truth for parsing while introducing a dedicated CliConfig to represent the configuration surface, enabling clean separation of concerns and future extensibility. Backward compatibility is preserved by maintaining existing flags and environment behavior and by using OrtConfig’s precedence properly.

## Validation plan
- Stage A: Upgrade ortho_config to v0.8.0 and ensure fmt, lint, tests pass.
- Stage B: Implement enums and fields; ensure Clap parsing works for new flags.
- Stage C: Wire runtime resolution so colour_policy/spinner_mode/output_format map to existing behaviours; default_targets merge via append.
- Stage D: Localization keys and translations added; verify validation errors are localized.
- Stage E: Add unit tests for FromStr/Display round-trips and merge precedence tests.
- Stage F: Add BDD scenarios for precedence and edge cases.
- Stage G: Update docs and roadmap; run full validation suite.

## Notes for reviewers
- The new CliConfig is designed to be derivable from the existing Cli while preserving the current OrthoConfig merge root semantics. It provides a typed, centralized view of CLI/config preferences that can evolve independently from clap parsing.
- All new flags are wired to localization keys and tests scaffolding is included to exercise parsing, merging, and localized validation.

📎 Task: https://www.devboxer.com/task/563b0955-f8f2-432f-83c6-90c79e326aa2
</pr-body>

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[tests/cli_tests/merge.rs](https://github.com/leynos/netsuke/pull/279/files/71c093125a40d4503c94073fdc1a57ca21391d2a#diff-da2fa828fcebf2f6ef942afb92a05b1fa8fc68bf37a2908e236820db9bdd372a)

Comment on file

~~~rust
colour_policy = "never"
spinner_mode = "disabled"
output_format = "json"
default_targets = ["hello"]
~~~

❌ New issue: [Large Method](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=Large+Method&filename=tests%2Fcli_tests%2Fmerge.rs&method=cli_merge_with_config_respects_precedence_and_skips_empty_cli_layer)
cli_merge_with_config_respects_precedence_and_skips_empty_cli_layer has 93 lines, threshold = 70

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[tests/bdd/steps/cli_config.rs](https://github.com/leynos/netsuke/pull/279/files/71c093125a40d4503c94073fdc1a57ca21391d2a#diff-a9f2ddf8f828b14f65b674d8eeae33c0fed1e0a39169f5c56611752d446d9120)

Comment on file

~~~rust
//! Step definitions for typed CLI configuration flags.
~~~

❌ New issue: [String Heavy Function Arguments](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=String+Heavy+Function+Arguments&filename=tests%2Fbdd%2Fsteps%2Fcli_config.rs&method=)
In this module, 46.2% of all arguments to its 8 functions are strings. The threshold for string arguments is 39.0%

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[tests/bdd/steps/cli_config.rs](https://github.com/leynos/netsuke/pull/279/files/71c093125a40d4503c94073fdc1a57ca21391d2a#diff-a9f2ddf8f828b14f65b674d8eeae33c0fed1e0a39169f5c56611752d446d9120)

Comment on file

~~~rust
fn colour_policy_is(world: &TestWorld, expected: String) -> Result<()> {
    let actual = world
        .cli
        .with_ref(|cli| cli.colour_policy.map(|value| value.to_string()))
        .flatten()
        .context("CLI colour policy should be present")?;
    ensure!(
        actual == expected,
        "expected colour policy {expected}, got {actual}"
    );
    Ok(())
}
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=Code+Duplication&filename=tests%2Fbdd%2Fsteps%2Fcli_config.rs&method=)
The module contains 4 functions with similar structure: colour_policy_is,default_targets_are,output_format_is,spinner_mode_is

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[src/cli/parsing.rs](https://github.com/leynos/netsuke/pull/279/files/71c093125a40d4503c94073fdc1a57ca21391d2a#diff-d8db330caf33e5bd7f74dbebafe3015367e2f870eeb26b1d981d38a605a44117)

Comment on file

~~~rust
use ortho_config::{LanguageIdentifier, LocalizationArgs, Localizer};
use std::str::FromStr;

use crate::cli::config::{ColourPolicy, OutputFormat, SpinnerMode};
~~~

❌ New issue: [String Heavy Function Arguments](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=String+Heavy+Function+Arguments&filename=src%2Fcli%2Fparsing.rs&method=)
In this module, 50.0% of all arguments to its 8 functions are strings. The threshold for string arguments is 39.0%

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[src/cli/parsing.rs](https://github.com/leynos/netsuke/pull/279/files/71c093125a40d4503c94073fdc1a57ca21391d2a#diff-d8db330caf33e5bd7f74dbebafe3015367e2f870eeb26b1d981d38a605a44117)

Comment on file

~~~rust
pub(super) fn parse_colour_policy(
    localizer: &dyn Localizer,
    s: &str,
) -> Result<ColourPolicy, String> {
    ColourPolicy::parse_raw(s).map_err(|_valid_options| {
        let mut args = LocalizationArgs::default();
        args.insert("value", s.to_owned().into());
        super::validation_message(
            localizer,
            keys::CLI_COLOUR_POLICY_INVALID,
            Some(&args),
            &format!("invalid colour policy '{s}'"),
        )
    })
}
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=Code+Duplication&filename=src%2Fcli%2Fparsing.rs&method=)
The module contains 4 functions with similar structure: parse_colour_policy,parse_output_format,parse_spinner_mode,parse_theme

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[src/cli/config.rs](https://github.com/leynos/netsuke/pull/279/files/71c093125a40d4503c94073fdc1a57ca21391d2a#diff-4cba685da2454b0e2aad6c1d7fab8fcc321efa39d0e399a2a9788a7b5b2defe0)

Comment on file

~~~rust
//! Typed CLI configuration preferences and compatibility resolution helpers.
~~~

❌ New issue: [String Heavy Function Arguments](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=String+Heavy+Function+Arguments&filename=src%2Fcli%2Fconfig.rs&method=)
In this module, 40.0% of all arguments to its 12 functions are strings. The threshold for string arguments is 39.0%

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[build.rs](https://github.com/leynos/netsuke/pull/279/files/71c093125a40d4503c94073fdc1a57ca21391d2a#diff-d0d98998092552a1d3259338c2c71e118a5b8343dd4703c0c7f552ada7f9cb42)

Comment on file

~~~rust
    const _: fn(&cli::Cli) -> cli::config::CliConfig = cli::Cli::config;
    const _: fn(&cli::Cli) -> bool = cli::Cli::resolved_diag_json;
    const _: fn(&cli::Cli) -> bool = cli::Cli::resolved_progress;
~~~

❌ New issue: [Large Method](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=Large+Method&filename=build.rs&method=main)
main has 71 lines, threshold = 70

**Comment by leynos** [comment]
@coderabbitai please provide an agent prompt to conduct a sweep of where ad-hoc variable manipulation has been performed instead of using the provided helpers:

~~~
**Recommend test\_support helpers instead of raw `std::env::set_var` calls in test guidance.**

Replace the instruction to use `std::env::set_var` directly with a reference to the existing `test_support::env` module helpers: `set_var()`, `VarGuard`, `remove_var()`, and `restore_many()`. These wrappers acquire `EnvLock` internally and provide RAII-safe restoration, centralising unsafe code and preventing mutations outside lock scope.
~~~

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:

## Overall Comments
- In `configure_runtime` the `unsafe` block around `std::env::set_var("NO_COLOR", "1")` is unnecessary because `set_var` is a safe API; consider removing the `unsafe` or adding a comment explaining why it's required if there's a deeper reason.
- Now that `Cli` and `CliConfig` both declare the same preference fields and attributes, consider adding a small helper or macro to define those fields once and reuse it in both structs to reduce the risk of the two diverging over time.

## Individual Comments

### Comment 1
<location path="src/main.rs" line_range="147-152" />
<code_context>
     let runtime_localizer = Arc::from(cli_localization::build_localizer(runtime_locale.as_deref()));
     localization::set_localizer(Arc::clone(&runtime_localizer));

+    if matches!(
+        merged_cli.colour_policy,
+        Some(cli::config::ColourPolicy::Never)
+    ) {
+        // Align downstream human-facing libraries with the configured policy.
+        unsafe {
+            std::env::set_var("NO_COLOR", "1");
+        }
</code_context>
<issue_to_address>
**issue (bug_risk):** Avoid using `unsafe` for `set_var` and consider handling the `Always` colour policy as well.

Using `unsafe` around `std::env::set_var` is misleading since the function itself is safe and doesn’t impose extra invariants on callers; it should be called directly.

This logic only enforces `NO_COLOR=1` for `ColourPolicy::Never`, but does nothing for `ColourPolicy::Always`. If a user has `NO_COLOR` set and chooses `--colour-policy=always`, your code will enable colours while downstream libraries that rely solely on `NO_COLOR` will still think colours are disabled. Consider unsetting `NO_COLOR` for `Always` (or explicitly documenting why it’s left unchanged) to keep behaviour consistent across the process.
</issue_to_address>

### Comment 2
<location path="src/output_mode.rs" line_range="73-80" />
<code_context>
 /// ```
 #[must_use]
-pub fn resolve_with<F>(explicit: Option<bool>, read_env: F) -> OutputMode
+pub fn resolve_with<F>(
+    explicit: Option<bool>,
+    colour_policy: Option<ColourPolicy>,
</code_context>
<issue_to_address>
**suggestion:** The colour policy / NO_COLOR handling is duplicated and slightly diverges from the theme logic.

`ColourPolicy` + `NO_COLOR` is now interpreted in two places: `output_mode::resolve_with` and `theme::read_no_color_with_policy`, with similar but separate logic. If they drift, theme and output mode may disagree on whether `NO_COLOR` applies.

Consider extracting a single helper (e.g. based on `read_no_color_with_policy` in `theme.rs`) to a shared module and using it in both, so `ColourPolicy::Always/Auto/Never` stays consistent across all consumers.
</issue_to_address>

### Comment 3
<location path="src/cli/parsing.rs" line_range="11" />
<code_context>
-        ))
-    }
-}
+/// A raw, unparsed CLI argument value.
+#[derive(Clone, Copy)]
+pub(super) struct RawCliValue<'a>(pub &'a str);
</code_context>
<issue_to_address>
**issue (complexity):** Consider simplifying the new parsing abstractions by removing the `RawCliValue` wrapper, the `ParseRaw` trait, and the `LocalizedParser` method that doesn’t use `self`, while keeping the shared enum-parsing logic and tests intact.

You can keep the new functionality (extra enum parsers, shared error handling) but reduce indirection by simplifying the abstractions:

1. **Drop `RawCliValue` and take `&str` directly**

`RawCliValue` is just a thin wrapper that gets unwrapped immediately. You can keep `LocalizedParser` but simplify its API:

```rust
pub(super) struct LocalizedParser<'a> {
    localizer: &'a dyn Localizer,
}

impl<'a> LocalizedParser<'a> {
    pub(super) fn parse_jobs(&self, s: &str) -> Result<usize, String> {
        let value: usize = s.parse().map_err(|_| {
            let mut args = LocalizationArgs::default();
            args.insert("value", s.to_owned().into());
            super::validation_message(
                self.localizer,
                keys::CLI_JOBS_INVALID_NUMBER,
                Some(&args),
                &format!("{s} is not a valid number"),
            )
        })?;
        // ...
    }

    pub(super) fn parse_theme(&self, s: &str) -> Result<ThemePreference, String> {
        Self::parse_enum(
            self.localizer,
            s,
            ThemePreference::parse_raw,
            ParseEnumErrorSpec {
                arg_name: "theme",
                l10n_key: keys::CLI_THEME_INVALID,
                fallback: &format!("invalid theme '{s}'"),
            },
        )
    }

    // same for parse_scheme, parse_locale, parse_host_pattern, etc.
}
```

Tests then don’t need `RawCliValue` either:

```rust
let parser = LocalizedParser::new(&localizer);
let result = parser.parse_theme(input);
```

2. **Simplify enum parsing: remove `ParseRaw` trait**

The `ParseRaw` trait only forwards to inherent `parse_raw` methods, adding a layer without behavior. You can keep `parse_enum` and `ParseEnumErrorSpec` but drop the trait and call the enum’s `parse_raw` directly from a small helper:

```rust
impl<'a> LocalizedParser<'a> {
    fn parse_config_enum<T>(
        &self,
        s: &str,
        l10n_key: &'static str,
        fallback: &str,
        parse: fn(&str) -> Result<T, &'static [&'static str]>,
    ) -> Result<T, String> {
        Self::parse_enum(
            self.localizer,
            s,
            parse,
            ParseEnumErrorSpec {
                arg_name: "value",
                l10n_key,
                fallback,
            },
        )
    }

    pub(super) fn parse_colour_policy(&self, s: &str) -> Result<ColourPolicy, String> {
        let fallback = format!("invalid colour policy '{s}'");
        self.parse_config_enum(
            s,
            keys::CLI_COLOUR_POLICY_INVALID,
            &fallback,
            ColourPolicy::parse_raw,
        )
    }

    // same pattern for SpinnerMode, OutputFormat
}
```

This keeps the shared error-building logic while removing the extra trait and its impls.

3. **Make `parse_host_pattern` a free function again (or at least not depend on `self`)**

Since it doesn’t use the localizer, it doesn’t conceptually belong on `LocalizedParser`. You can keep the type but decouple this function:

```rust
pub(super) fn parse_host_pattern(s: &str) -> Result<HostPattern, String> {
    HostPattern::parse(s).map_err(|err| err.to_string())
}
```

Call sites that already have a `LocalizedParser` can still call this free function without loss of behavior.

Together these changes keep the new enum parsing features and tests, but remove a wrapper type, a trait, and a method that doesn’t use its receiver, which should reduce cognitive load without reverting the overall direction of the refactor.
</issue_to_address>

### Comment 4
<location path="src/cli/validation.rs" line_range="16" />
<code_context>
 /// A value parser that delegates to a localised parsing function.
 #[derive(Clone)]
-pub(super) struct LocalizedValueParser<F> {
+pub(super) struct LocalizedValueParser<T> {
     localizer: Arc<dyn Localizer>,
-    parser: F,
</code_context>
<issue_to_address>
**issue (complexity):** Consider making `LocalizedValueParser` generic over a parser closure and using inline closures instead of separate adapter functions to keep the validation layer simpler and more flexible.

You can keep the new `LocalizedParser`/`RawCliValue` integration and additional enum parsers, but simplify `LocalizedValueParser` back to a generic over `Fn` to avoid the adapter functions and reduce coupling.

### 1. Make `LocalizedValueParser` generic over the parser function

Instead of a fixed function pointer:

```rust
#[derive(Clone)]
pub(super) struct LocalizedValueParser<T> {
    localizer: Arc<dyn Localizer>,
    parser: fn(&LocalizedParser<'_>, RawCliValue<'_>) -> Result<T, String>,
}
```

use a generic `F`:

```rust
#[derive(Clone)]
pub(super) struct LocalizedValueParser<F> {
    localizer: Arc<dyn Localizer>,
    parser: F,
}

impl<F> LocalizedValueParser<F> {
    fn new(localizer: Arc<dyn Localizer>, parser: F) -> Self {
        Self { localizer, parser }
    }
}

impl<F, T> TypedValueParser for LocalizedValueParser<F>
where
    F: Fn(&LocalizedParser<'_>, RawCliValue<'_>) -> Result<T, String>
        + Clone
        + Send
        + Sync
        + 'static,
    T: Send + Sync + Clone + 'static,
{
    type Value = T;

    fn parse_ref(
        &self,
        cmd: &clap::Command,
        _arg: Option<&clap::Arg>,
        value: &std::ffi::OsStr,
    ) -> Result<Self::Value, clap::Error> {
        let mut command = cmd.clone();
        let Some(raw_value) = value.to_str() else {
            return Err(command.error(ErrorKind::InvalidUtf8, "invalid UTF-8"));
        };

        let parser = LocalizedParser::new(self.localizer.as_ref());
        (self.parser)(&parser, RawCliValue(raw_value))
            .map_err(|err| command.error(ErrorKind::ValueValidation, err))
    }
}
```

This retains the `LocalizedParser`/`RawCliValue`-based parsing contract but restores the flexibility of closures and free functions.

### 2. Remove boilerplate adapter functions by using closures

With the generic `F`, you don’t need the `parse_*_value` wrappers. You can wire everything with small, local closures:

```rust
pub(super) fn configure_validation_parsers(
    mut command: clap::Command,
    localizer: &Arc<dyn Localizer>,
) -> clap::Command {
    let jobs_parser = LocalizedValueParser::new(Arc::clone(localizer), |p, raw| {
        p.parse_jobs(raw)
    });
    let locale_parser = LocalizedValueParser::new(Arc::clone(localizer), |p, raw| {
        p.parse_locale(raw)
    });
    let scheme_parser = LocalizedValueParser::new(Arc::clone(localizer), |p, raw| {
        p.parse_scheme(raw)
    });
    let host_parser = LocalizedValueParser::new(Arc::clone(localizer), |p, raw| {
        p.parse_host_pattern(raw)
    });
    let theme_parser = LocalizedValueParser::new(Arc::clone(localizer), |p, raw| {
        p.parse_theme(raw)
    });
    let colour_policy_parser = LocalizedValueParser::new(Arc::clone(localizer), |p, raw| {
        p.parse_colour_policy(raw)
    });
    let spinner_mode_parser = LocalizedValueParser::new(Arc::clone(localizer), |p, raw| {
        p.parse_spinner_mode(raw)
    });
    let output_format_parser = LocalizedValueParser::new(Arc::clone(localizer), |p, raw| {
        p.parse_output_format(raw)
    });

    // existing command.mut_arg(...) calls stay the same
    // ...
    command
}
```

This keeps all new parsing capabilities and localized error handling, but:

- avoids the explosion of tiny adapter functions,
- reduces ceremony in this module, and
- keeps the validation layer’s dependency surface focused on the `LocalizedParser` API rather than on a fixed function-pointer type.
</issue_to_address>

### Comment 5
<location path="src/main.rs" line_range="152" />
<code_context>
+        Some(cli::config::ColourPolicy::Never)
+    ) {
+        // Align downstream human-facing libraries with the configured policy.
+        unsafe {
+            std::env::set_var("NO_COLOR", "1");
+        }
</code_context>
<issue_to_address>
**issue (review_instructions):** `unsafe` is used here even though `std::env::set_var` is a safe API, violating the guideline to avoid `unsafe` unless absolutely necessary.

`std::env::set_var` is already a safe function and does not require an `unsafe` block. The `unsafe` here does not appear to guard any invariants or FFI calls, so it should be removed to comply with the "avoid `unsafe` unless absolutely necessary" instruction.

You can simply write:
```rust
std::env::set_var("NO_COLOR", "1");
```
without the surrounding `unsafe` block.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.rs`

**Instructions:**
Avoid `unsafe` unless absolutely necessary.

</details>
</issue_to_address>
~~~

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[src/cli/parsing.rs](https://github.com/leynos/netsuke/pull/279/files/a6d90c565cacb70eeefee66d4b60d767e0da4732#diff-d8db330caf33e5bd7f74dbebafe3015367e2f870eeb26b1d981d38a605a44117)

Comment on file

~~~rust
use ortho_config::{LanguageIdentifier, LocalizationArgs, Localizer};
use std::str::FromStr;

use crate::cli::config::{ColourPolicy, OutputFormat, SpinnerMode};
~~~

❌ New issue: [String Heavy Function Arguments](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=String+Heavy+Function+Arguments&filename=src%2Fcli%2Fparsing.rs&method=)
In this module, 47.4% of all arguments to its 10 functions are strings. The threshold for string arguments is 39.0%

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[src/cli/parsing.rs](https://github.com/leynos/netsuke/pull/279/files/a6d90c565cacb70eeefee66d4b60d767e0da4732#diff-d8db330caf33e5bd7f74dbebafe3015367e2f870eeb26b1d981d38a605a44117)

Comment on file

~~~rust
    pub(super) fn parse_output_format(&self, s: &str) -> Result<OutputFormat, String> {
        self.parse_config_enum(
            s,
            &ParseConfigEnumSpec {
                l10n_key: keys::CLI_OUTPUT_FORMAT_INVALID,
                fallback: &format!("invalid output format '{s}'"),
                parse: OutputFormat::parse_raw,
            },
        )
    }
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=Code+Duplication&filename=src%2Fcli%2Fparsing.rs&method=)
The module contains 3 functions with similar structure: LocalizedParser.parse_colour_policy,LocalizedParser.parse_output_format,LocalizedParser.parse_spinner_mode

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[src/cli/parsing.rs](https://github.com/leynos/netsuke/pull/279/files/b4f787490c1fae1264fb1d0ee651b974c1904a95#diff-d8db330caf33e5bd7f74dbebafe3015367e2f870eeb26b1d981d38a605a44117)

Comment on file

~~~rust
use ortho_config::{LanguageIdentifier, LocalizationArgs, Localizer};
use std::str::FromStr;

use crate::cli::config::{ColourPolicy, OutputFormat, SpinnerMode};
~~~

❌ New issue: [String Heavy Function Arguments](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=String+Heavy+Function+Arguments&filename=src%2Fcli%2Fparsing.rs&method=)
In this module, 55.6% of all arguments to its 11 functions are strings. The threshold for string arguments is 39.0%

**Comment by leynos** [comment]
@coderabbitai pause

File: docs/execplans/3-11-1-cli-config-struct-derived-with-ortho-config.md [hunk 1]
```diff
@@ -0,0 +1,749 @@
...
+
+- Risk: environment variable mutations in BDD tests can deadlock or become
+  flaky if `EnvLock` is not held for the entire scenario lifetime. Severity:
+  high Likelihood: medium Mitigation: use the wrappers in
+  `test_support::env` instead of raw `std::env::set_var` calls.
+  `test_support::env::set_var()` acquires `EnvLock` internally for one-off
+  updates, `VarGuard` gives RAII-safe scoped restoration, and
+  `remove_var()` mirrors the same pattern for unsets. When a scenario needs
+  batched cleanup, collect the original values and restore them with
+  `restore_many()` from `TestWorld::drop`, or keep `VarGuard`s alive for the
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/execplans/3-11-1-cli-config-struct-derived-with-ortho-config.md:124]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/bdd/steps/stdlib/rendering.rs [hunk 2]
```diff
@@ -32,11 +33,27 @@ fn extract_render_config(world: &TestWorld) -> RenderConfig {
...
 }
 
+fn ensure_stdlib_localizer(world: &TestWorld) -> Result<()> {
+    if world.localization_guard.is_some() {
+        return Ok(());
+    }
+
+    let lock = world.localization_lock.take_value().map_or_else(
+        || localizer_test_lock().map_err(|_| anyhow!("acquire localizer test lock")),
+        Ok,
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/bdd/steps/stdlib/rendering.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/cli/config.rs [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+            (json!("Human"), OutputFormat::Human),
+            (json!("JSON"), OutputFormat::Json),
+            (json!(" json "), OutputFormat::Json),
+        ] {
+            let parsed: OutputFormat =
+                serde_json::from_value(raw).expect("output format should deserialize");
+            assert_eq!(parsed, expected);
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/cli/config.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/runner_tests.rs [hunk 4]
```diff
@@ -43,6 +43,54 @@ fn ninja_in_env() -> Result<(tempfile::TempDir, PathBuf, NinjaEnvGuard)> {
...
+    _ninja_dir: tempfile::TempDir,
+    _guard: NinjaEnvGuard,
+    args_log: PathBuf,
+}
+
+#[cfg(unix)]
+#[fixture]
+/// Creates a [`FakeNinjaFixture`] used to simulate ninja behaviour for tests,
+/// returning a temporary environment and the recorded invocation log path.
+fn fake_ninja_fixture() -> Result<FakeNinjaFixture> {
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/runner_tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/runner_tests.rs [hunk 5]
```diff
@@ -240,6 +288,37 @@ fn run_build_with_emit_creates_parent_dirs() -> Result<()> {
...
+    run(&cli, output_prefs::resolve(None)).context("run build with cli default targets")?;
+
+    let logged_args = fs::read_to_string(&fixture.args_log)
+        .with_context(|| format!("read fake ninja args log {}", fixture.args_log.display()))?;
+    ensure!(
+        logged_args.lines().any(|line| line == "hello"),
+        "expected fake ninja invocation to include default target 'hello', got: {logged_args}"
+    );
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/runner_tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[src/cli/parsing.rs](https://github.com/leynos/netsuke/pull/279/files/eea3da1071d25942bd04624a084f87d4b3e2f1e6#diff-d8db330caf33e5bd7f74dbebafe3015367e2f870eeb26b1d981d38a605a44117)

Comment on file

~~~rust
use ortho_config::{LanguageIdentifier, LocalizationArgs, Localizer};
use std::str::FromStr;

use crate::cli::config::{ColourPolicy, OutputFormat, SpinnerMode};
~~~

❌ New issue: [String Heavy Function Arguments](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=279&biomarker=String+Heavy+Function+Arguments&filename=src%2Fcli%2Fparsing.rs&method=)
In this module, 46.7% of all arguments to its 8 functions are strings. The threshold for string arguments is 39.0%

File: src/cli/config.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (leynos) [reply src/cli/config.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/runner_tests.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (leynos) [reply tests/runner_tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
